js/opentok-hardware-setup.js

```diff
@@ -286,7 +286,16 @@ var authenticateForDeviceLabels = function(callback) {
             return callback(new Error('getUserMedia not supported in this browser'));
           }
           getUserMedia(constraints, function(stream) {
-            stream.stop();
+            if (stream.stop) {
+                // Older spec
+                stream.stop();
+            } else {
+                // Newer spec
+                var tracks = stream.getAudioTracks().concat(stream.getVideoTracks());
```

Comment:
You may want to test for the newer spec version first. This is avoid warnings in the js console. E.g.

``` javascript
if (window.MediaStreamTrack && window.MediaStreamTrack.prototype.stop) {
  // newer spec
} else if (stream.stop) {
 // older spec
} else {
 // WUT? Either log it (if appropriate) or remove the else branch
}
```